docs/build/docs/content/libmav.md

```diff
@@ -0,0 +1,1364 @@
+- `heartbeat_message` _Message_ - The HEARTBEAT message to stream.
+- `interface` _NetworkInterface_ - An object defining the physical connection that is being watched for MAVLink traffic.
+  This is one of [UDPClient](#libmav.UDPClient), [UDPServer](#libmav.UDPServer), [TCPClient](#libmav.TCPClient), [TCPServer](#libmav.TCPServer), [Serial](#libmav.Serial).
+  
+
+**Returns**:
+
+- `NetworkRuntime` - A new network runtime object that can be waited on for a connection.
+
+<a id="libmav.NetworkRuntime.await_connection"></a>
+
+#### await\_connection
+
+```python
+def await_connection(timeout)
+```
+
+Wait for the remote end of network to connect, or timeout.
+
+Connection is defined as "detecting MAVLink messages". <!-- any messages or heartbeat? What about disconnection? -->
+
+
+```python
+# Connect to a TCP server
+conn_physical = libmav.TCPClient('192.168.1.12', 14550)
+conn_runtime = libmav.NetworkRuntime(self.message_set, heartbeat, conn_physical)
+
+# Return Connection or throw Exception
+connection = conn_runtime.await_connection(2000)
+```
+
+**Arguments**:
+
+- `timeout` _int_ - Time to wait for connection before raising exception, in milliseconds
+  
+
+**Returns**:
+
+- `Connection` - An object representing a connection to the MAVLink network.
+
+<a id="libmav.NetworkRuntime.clear_heartbeat_message"></a>
+
+#### clear\_heartbeat\_message
+
+```python
+def clear_heartbeat_message()
+```
+
+Clear the heartbeat message (if set).
+
+This stops streaming the heartbeat message, which may have been set in the class initializer or using `set_heartbeat_message()`.
+
+```python
+connection.clear_heartbeat_message()
+```
+
+<a id="libmav.NetworkRuntime.on_connection"></a>
+
+#### on\_connection
+
+```python
+def on_connection(callbackFn)
+```
+
+Set a callback function that is called whenever a new `Connection` is detected.
+
+A Connection is created if MAVLink traffic is detected from a unique remote address/port.
+
+The callback function is called with the `Connection` object.
+The identify of the unique `ConnectionPartner` associated with the `Connection` be determined using the `Connection.partner()` method.
+Note that a `NetworkRuntime` constructed using a `TCPClient`, `UDPClient` or `Serial` network interface will only have one connection (because there is only one partner), while a `TCPServer` or `UDPServer` might have several remote systems connecting from different IP addresses.
+
+
+
+```python
+# Define a callback function
+def callbackConnected(connection):
+    print(f'New connection: {connection}')
+
+conn_runtime.on_connection(callbackConnected)        
+ ```
+
+**Arguments**:
+
+- `callbackFn` _method_ - Callback function that takes a `Connection` object as its argument.
+
+<a id="libmav.NetworkRuntime.on_connection_lost"></a>
+
+#### on\_connection\_lost
+
+```python
+def on_connection_lost()
+```
+
+Set a callback function that is called whenever a `Connection` is lost.
```

Comment:
Here I still think libmav is not veyr good. As of now, whenever a connection is lost, you are basially forced to recreate the `Interface`, the `NetworkRuntime` and then wait for the `Connection` again. 

I have a PR on the C++ side to make connections revivable, which should improve things.

Reply:
Yes. I'll add a note for how it is now to the source